packages/integration-karma/test/accessibility/LightningElement.focus/index.spec.js

```diff
@@ -28,70 +27,73 @@ if (!process.env.NATIVE_SHADOW) {
-        const elm = createElement('x-focus', { is: Ctor });
-        document.body.appendChild(elm);
+    it('should focus the host element (delegatesFocus=false, tabIndex=-1)', () => {
+        const container = createElement('x-container', { is: Container });
+        document.body.appendChild(container);
 
-        const input = elm.shadowRoot.querySelector('.second');
-        input.focus();
-        input.blur();
+        const elm = container.shadowRoot.querySelector('x-delegates-focus-false[tabindex="-1"]');
+        elm.focus();
+        expect(container.shadowRoot.activeElement).toBe(elm);
         expect(elm.shadowRoot.activeElement).toBeNull();
     });
 
-    it(`should not move focus if an internal element is already focused ${category}`, () => {
-        const elm = createElement('x-focus', { is: Ctor });
-        document.body.appendChild(elm);
+    it('should focus the host element (delegatesFocus=false, tabIndex=0)', () => {
+        const container = createElement('x-container', { is: Container });
+        document.body.appendChild(container);
 
-        const input = elm.shadowRoot.querySelector('.second');
-        input.focus();
+        const elm = container.shadowRoot.querySelector('x-delegates-focus-false[tabindex="0"]');
         elm.focus();
-        expect(elm.shadowRoot.activeElement).toBe(input);
+        expect(container.shadowRoot.activeElement).toBe(elm);
+        expect(elm.shadowRoot.activeElement).toBeNull();
     });
-}
 
-runFocusTests({ delegatesFocus: true });
-runFocusTests({ delegatesFocus: false });
+    // Tests that apply regardless of whether focus is being delegated
+    function runFocusTests({ delegatesFocus }) {
+        const category = `(delegatesFocus=${delegatesFocus})`;
+        const Ctor = delegatesFocus ? DelegatesFocusTrue : DelegatesFocusFalse;
+
+        it(`should focus the internal element when invoking the focus method directly on the internal element ${category}`, () => {
+            const elm = createElement('x-focus', { is: Ctor });
+            document.body.appendChild(elm);
+
+            const input = elm.shadowRoot.querySelector('.second');
+            input.focus();
+            expect(elm.shadowRoot.activeElement).toBe(input);
+        });
+
+        it(`should blur the internal element when invoking the blur method directly on the internal element ${category}`, () => {
+            const elm = createElement('x-focus', { is: Ctor });
+            document.body.appendChild(elm);
+
+            const input = elm.shadowRoot.querySelector('.second');
+            input.focus();
+            input.blur();
+            expect(elm.shadowRoot.activeElement).toBeNull();
+        });
+
+        // TODO [#2566]: Firefox behaves differently from Chrome/Safari in this test
+        if (!(process.env.NATIVE_SHADOW && delegatesFocus && window.mozInnerScreenX)) {
```

Comment:
I'm typically not a fan of UA sniffing, but it seems better than completely disabling the test.